### PR TITLE
Add CLI to known-good for 2.1 RC

### PR DESCRIPTION
**Pull request by dseefeld**

### CODE REVIEW

File: patches/cli/0012-Exclude-Localization-in-source-build.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/src/redist/redist.csproj b/src/redist/redist.csproj
+index da07741..c2b2a87 100644
+--- a/src/redist/redist.csproj
++++ b/src/redist/redist.csproj
+@@ -19,7 +19,7 @@
+     <PackageReference Include="NuGet.Build.Tasks" Version="$(NuGetBuildTasksPackageVersion)" />
+     <PackageReference Include="Microsoft.TestPlatform.CLI" Version="$(MicrosoftTestPlatformCLIPackageVersion)" />
+     <PackageReference Include="Microsoft.TestPlatform.Build" Version="$(MicrosoftTestPlatformBuildPackageVersion)" />
+-    <PackageReference Include="NuGet.Localization" Version="$(NuGetProjectModelPackageVersion)" />
++    <PackageReference Condition=" '$(DotNetBuildFromSource)' != 'true' " Include="NuGet.Localization" Version="$(NuGetProjectModelPackageVersion)" />
```
Comment (dagood) [inline patches/cli/0012-Exclude-Localization-in-source-build.patch]: It looks like this package contains a bunch of string loc resource DLLs. Any idea if CLI commands might look for these translations and fail if we don't include them? Think we need to do a non-English smoke test?

Comment (dagood) [reply patches/cli/0012-Exclude-Localization-in-source-build.patch]: I've never tried it before, not sure what you have to do.

Comment (dagood) [reply patches/cli/0012-Exclude-Localization-in-source-build.patch]: The locale-named folders in the CLI we download and the preview2 tarball-built CLI are all empty, so it seems normal to me for these not to exist. I'm guessing we're probably fine here, but I filed https://github.com/dotnet/source-build/issues/485 to follow up more generally on localization.

Comment (crummel) [reply patches/cli/0012-Exclude-Localization-in-source-build.patch]: My Debian machine has the official dotnet build in German now so I can test this out manually at least.  It looks like switching locales does not work in a Docker container, and you have to remove locale variables from /etc/ssh/sshd_config so they aren't overridden when you log in.

File: repos/cli.proj [hunk 2]
```diff
@@ -48,8 +48,8 @@
     <RepositoryReference Include="newtonsoft-json" />
     <RepositoryReference Include="nuget-client" />
     <RepositoryReference Include="roslyn" />
-    <RepositoryReference Include="sdk" />
-    <RepositoryReference Include="templating" />
+    <!--RepositoryReference Include="sdk" /-->
+    <!--RepositoryReference Include="templating" /-->
```
Comment (crummel) [inline repos/cli.proj]: I think templating should work now, I've got this + Davis' changes to enable cli-migrate and sdk building locally so I will update if I run into problems.

**Comment by crummel** [comment]
RHEL tarball error in case the build goes away:
```
"/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/cli.proj" (Build target) (5:2) ->
(RepoBuild target) -> 
  2>CROSSGEN : error : Could not load file or assembly 'System.Security.Cryptography.ProtectedData, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/cli/src/redist/redist.csproj] [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/cli.proj]
  2>CROSSGEN : error : compilation failed for "/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/cli/bin/2/linux-x64/dotnet/sdk/2.1.300-rc1-008669/NuGet.Configuration.dll" (0x80131621) [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/cli/src/redist/redist.csproj] [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/cli.proj]
  CROSSGEN : error : Could not load file or assembly 'System.Security.Cryptography.ProtectedData, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/cli/src/redist/redist.csproj] [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/cli.proj]
  CROSSGEN : error : compilation failed for "/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/cli/bin/2/linux-x64/dotnet/sdk/2.1.300-rc1-008669/NuGet.Configuration.dll" (0x80131621) [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/cli/src/redist/redist.csproj] [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/cli.proj]
  /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/dir.targets(187,5): error MSB3073: The command "/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/src/cli/build.sh --stage0 /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/Tools/dotnetcli/ /p:NETCoreAppMaximumVersion=99.9 /p:CoreSetupBlobRootUrl=file:///mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/bin/obj/x64/Release/blobs/ /p:SkipBuildingInstallers=true /p:IncludeNuGetPackageArchive=false /p:IncludeAdditionalSharedFrameworks=false /p:UsePortableLinuxSharedFramework=false /p:IncludeSharedFrameworksForBackwardsCompatibilityTests=false /p:IncludeAspNetCoreRuntime=false /p:IncludeMSBuildSdkResolver=false '/p:CLITargets="Prepare;Compile;Package"' /v:detailed  /p:DotNetPackageVersionPropsPath=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/bin/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/bin/obj/x64/Release/RestoreSources.props /p:DotNetBuildOffline=true " exited with code 1. [/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.1/RHEL7.2_Tarball_Release_prtest/tarball-output/repos/cli.proj]
```

**Comment by dagood** [comment]
Per https://github.com/dotnet/source-build/issues/487, I removed the ProtectedData reference assembly package to let the NuGet.Client build pick up the prebuilt instead.

**Comment by dagood** [comment]
My local offline tarball build succeeded. The smoke test doesn't work because the package version doesn't line up with what CoreCLR source-built: `Unable to find package Microsoft.NETCore.Platforms with version (>= 2.1.0-rc1-26419-03)`, vs. built `Microsoft.NETCore.Platforms.2.1.0-rc1-26406-04.nupkg`.

**Pull request merged**